index.js

```diff
@@ -63,34 +113,123 @@ module.exports = () => {
 
     selectIpfsReady: state => state.ipfs.ready,
 
-    selectIpfsApiAddress: state => state.ipfs.apiAddress,
+    selectIpfsProvider: state => state.ipfs.provider,
+
+    selectIpfsApiOpts: state => state.ipfs.apiOpts,
 
-    selectIpfsInitFailed: state => !!state.ipfs.error,
+    selectIpfsInitFailed: state => state.ipfs.failed,
 
     selectIpfsIdentity: state => state.ipfs.identity,
 
-    doInitIpfs: () => async ({ dispatch, getState }) => {
-      dispatch({ type: 'IPFS_INIT_STARTED' })
+    doInitIpfs: () => async (store) => {
+      await getIpfs(opts, store)
+    },
 
-      const { apiAddress } = getState().ipfs
-      let identity = null
+    doUpdateIpfsApiOpts: (usrOpts) => (store) => {
+      store.dispatch({ type: 'IPFS_API_UPDATED', payload: usrOpts })
+      saveUserOpts('ipfsApi', usrOpts)
```

Comment:
Actions tell us that something just happened, so the call to saveUserOpts should come before the dispatch, as the api isn't really updated until it's been stored.

Now we're dealing with an api opts object the action would be more accurately called `IPFS_API_OPTS_UPDATED`